### PR TITLE
NEWS for PR #4452

### PR DESCRIPTION
**Pull request by mitza-oci**

### CODE REVIEW

File: docs/news.d/msvc-stdcpp.rst [hunk 1]
```diff
@@ -0,0 +1,5 @@
+.. news-prs: 4452
+
+.. news-start-section: Additions
+- When using Visual C++, OpenDDS can now be configured using -std=c++NN (NN = 17 or 20).
```
Comment (iguessthislldo) [inline docs/news.d/msvc-stdcpp.rst:4]:
```suggestion
.. news-start-section: Platform Support and Dependencies
- When using Visual C++, OpenDDS can now be configured using ``--std=c++NN`` (NN = 17 or 20).
```

**Review by iguessthislldo — CHANGES_REQUESTED** [review]

File: docs/news.d/msvc-stdcpp.rst (hunk 1 above)
Comment (mitza-oci) [reply docs/news.d/msvc-stdcpp.rst]: @iguessthislldo this didn't work

```
KeyError: "Can't define a new main section named 'Platform Support and Dependencies'
```

**Review by iguessthislldo — APPROVED** [review]

File: docs/news.d/msvc-stdcpp.rst (hunk 1 above)
Comment (iguessthislldo) [reply docs/news.d/msvc-stdcpp.rst]: Ah, I documented it as a main section:

https://github.com/OpenDDS/OpenDDS/blob/932dbb884da6434564c4ba07c35313ce61c7aef5/docs/internal/docs.rst?plain=1#L544-L545

but the news script doesn't have it as a main section:

https://github.com/OpenDDS/OpenDDS/blob/932dbb884da6434564c4ba07c35313ce61c7aef5/docs/sphinx_extensions/newsd.py#L13-L22

Guess we haven't tried to use it until now.

Comment (iguessthislldo) [reply docs/news.d/msvc-stdcpp.rst]: So the sections do need to be builtin so they have a builtin order, but I'm now wondering if it should reject unknown top level sections like this. Regardless, could you add the section for me? It's below Additions in the documentation, so maybe:

```python
 main_sections = { 
     'Additions': 100, 
     'Platform Support and Dependencies': 95,
     'Deprecations': 90, 
     'Removals': 80, 
     'Security': 60, 
     'Fixes': 40, 
     'Documentation': 20, 
     'Notes': 0, 
 } 
```

**Review by iguessthislldo — APPROVED** [review]

**Pull request merged**